test/interaction_events.test.js

```diff
@@ -0,0 +1,976 @@
+// These tests ensure that user interactions fire the right events
+
+import test from 'tape';
+import spy from 'sinon/lib/sinon/spy'; // avoid babel-register-related error by importing only spy
+import createSyntheticEvent from 'synthetic-dom-events';
+import GLDraw from '../';
+import {
+  createMap,
+  click
+} from './test_utils';
+import createAfterNextRender from './utils/after_next_render';
+import makeMouseEvent from './utils/make_mouse_event';
+
+const container = document.createElement('div');
+document.body.appendChild(container);
+const map = createMap({ container });
+const fireSpy = spy(map, 'fire');
+const afterNextRender = createAfterNextRender(map);
+const Draw = GLDraw();
+map.addControl(Draw);
+
+map.on('load', runTests);
+document.body.removeChild(container);
+
+const backspaceEvent = createSyntheticEvent('keydown', {
+  keyCode: 8
+});
+const enterEvent = createSyntheticEvent('keyup', {
+  keyCode: 13
+});
+const escapeEvent = createSyntheticEvent('keyup', {
+  keyCode: 27
+});
+
+function runTests() {
+  const pointButton = container.getElementsByClassName('mapbox-gl-draw_point')[0];
+  const lineButton = container.getElementsByClassName('mapbox-gl-draw_line')[0];
+  const trashButton = container.getElementsByClassName('mapbox-gl-draw_trash')[0];
+  const polygonButton = container.getElementsByClassName('mapbox-gl-draw_polygon')[0];
+
+  // The sequence of these tests matters: each uses state established
+  // in the prior tests. These variables keep track of bits of that state.
+
+  test('enter draw_point mode', t => {
+    fireSpy.reset();
+
+    // Click the line button
+    pointButton.click();
+    firedWith(t, 'draw.modechange', {
+      mode: 'draw_point'
+    });
+    t.deepEqual(flushDrawEvents(), ['draw.modechange'], 'no unexpected draw events');
+    t.end();
+  });
+
+  const point25_25GeoJson = {
+    type: 'Feature',
+    properties: {},
+    geometry: {
+      type: 'Point',
+      coordinates: [25, 25]
+    }
+  };
+
+  test('create a point', t => {
+    // Now in `draw_point` mode ...
+    // Move around, then click to create the point
+    map.fire('mousemove', makeMouseEvent(10, 10));
+    map.fire('mousemove', makeMouseEvent(20, 20));
+    click(map, makeMouseEvent(25, 25));
+    afterNextRender(() => {
+      const pointId = firedWith(t, 'draw.create', {
+        features: [point25_25GeoJson]
+      }).features[0].id;
+
+      firedWith(t, 'draw.modechange', {
+        mode: 'simple_select',
+        options: { featureIds: [pointId] }
```

Comment:
So, I think `draw.selectionchange` does what `opts` used to do. I think that makes sense. Even if we wanted to expose what Position has been selected I think we'd want to do it in `draw.selectionchange`